### ISSUE TITLE
SIMD: Implement batch activate_and_trace for backpropagation

### ISSUE THREAD

**Issue by stservice** [stSoftwareAU/NEAT-TS#1212]
## Background

The current \`activate_and_trace()\` function (line ~3249 in lib.rs) processes a single record at a time, storing activation values and pre-squash hint values for backpropagation. This is a significant bottleneck during training when processing mini-batches.

## Current State

```rust
pub fn activate_and_trace(
    &mut self,
    inputs: &[f32],
    num_outputs: usize,
) -> Vec<f32> {
    // Single record processing
    // Traces hint values for gradient computation
    // Returns [outputs..., activations..., hints..., trace_data...]
}
```

When training with mini-batches (default \`batchSize: 64\`), this is called 64 times sequentially before gradients are computed.

## Proposal

Implement \`activate_and_trace_batch_4way()\` that processes 4 records simultaneously:

```rust
pub fn activate_and_trace_batch_4way(
    &self,
    inputs: &[f32],  // packed: [input0..., input1..., input2..., input3...]
    input_size: usize,
    num_outputs: usize,
) -> (Vec<f32>, Vec<f32>, Vec<f32>, Vec<f32>) {
    // Returns 4 trace results, one per record
    // Each contains: [outputs..., activations..., hints..., trace_data...]
}
```

## Expected Benefits

- **4x speedup** for backpropagation setup phase
- Enables parallel gradient accumulation
- Better cache utilisation when processing mini-batches
- Prerequisite for SIMD derivative/gradient computation

## Implementation Notes

1. Reuse the \`weighted_sum_simd_4records()\` infrastructure from PR #1205
2. Allocate 4 sets of activation and hint buffers
3. Process all neurons for 4 records in parallel
4. Store pre-squash values (hints) for all 4 records
5. Return packed or separate result buffers

## Integration Points

- \`src/propagate/BackPropagation.ts\`: Could batch calls to activate_and_trace
- Training loops that use \`batchSize\` configuration
- Would pair well with batched derivative computation (see #1211)

## Files to Modify

- \`wasm_activation/src/lib.rs\`: Add \`activate_and_trace_batch_4way()\`
- Potentially \`src/propagate/BackPropagation.ts\` for integration

## Success Criteria

- 4x speedup for trace generation in benchmarks
- All backpropagation tests pass
- Training convergence unaffected
- Numerical precision matches single-record implementation

Related to: #1205, #1202

**Comment by stservice** [comment]
## Automated Processing Failed (First Attempt)

The automated worker was unable to complete this issue on the first attempt.

### Failure Details
Claude timed out after 3600 seconds (60 minutes).

This issue may be too complex, too large, or require additional permissions that Claude doesn't have.

<details>
<summary>Last output from Claude (click to expand)</summary>

```
No output captured
```
</details>

### What happens next?
- This issue has been marked with the `failed-once` label
- The worker will **automatically retry** this issue on the next scan
- If it fails again, the issue will be marked as `failed` and require manual intervention

### To prevent automatic retry
If you want to work on this manually, remove the `failed-once` label and assign yourself.

**Comment by stservice** [comment]
## Automated Processing Failed (First Attempt)

The automated worker was unable to complete this issue on the first attempt.

### Failure Details
Claude timed out after 3600 seconds (60 minutes).

This issue may be too complex, too large, or require additional permissions that Claude doesn't have.

<details>
<summary>Last output from Claude (click to expand)</summary>

```
No output captured
```
</details>

### What happens next?
- This issue has been marked with the `failed-once` label
- The worker will **automatically retry** this issue on the next scan
- If it fails again, the issue will be marked as `failed` and require manual intervention

### To prevent automatic retry
If you want to work on this manually, remove the `failed-once` label and assign yourself.
